Sources/Main.cpp

```diff
@@ -1136,7 +1147,12 @@ void __fastcall TX584Form::DeleteItemClick(TObject *Sender)
             SelCount = 1;
         } else
             //очищаем выделенные инструкции
-            for (int i = CodeListView->ItemIndex; i < CodeListView->ItemIndex + SelCount; i++) {
+            for (int i = SelStart;
```

Comment:
omg, this is messy :exploding_head:  Please never write such code at work unless you want the first prize at code obfuscation contest. Try to find a better approach or at least refactor complex expressions into simpler ones using helper variables and methods. For example, this code can be rewritten using the suggested helper method `GetSelection`:

```cpp
int SelStart, SelEnd;
GetSelection(SelStart, SelEnd);
for (int i = SelStart; i < SelEnd; i++) {
```

Reply:
Please don't forget to fix deletion of selected items when INSERT mode is active. It also uses that pesky `CodeListView->ItemIndex`.

Reply:
I tried to introduce some helper variables but compiler complained that they're not exist so I wrote this obfuscated code.

Reply:
Hmm, that's strange. Perhaps the compiler was confused because of missing curly braces:

```cpp
} else
    //очищаем выделенные инструкции
    int MyHelperVariable = CodeListView->ItemIndex;
    for (int i = MyHelperVariable; i < MyHelperVariable + SelCount; i++) {
        Code[i] = NOP;
        CodeListView->Items->Item[i]->SubItems->Strings[1] = NOP_TEXT;
        CodeListView->Items->Item[i]->SubItems->Strings[2] = "";
    }
```

The compiler interprets this as following:

```cpp
} else {
    //очищаем выделенные инструкции
    int MyHelperVariable = CodeListView->ItemIndex;
}

for (int i = MyHelperVariable; i < MyHelperVariable + SelCount; i++) {
    Code[i] = NOP;
    CodeListView->Items->Item[i]->SubItems->Strings[1] = NOP_TEXT;
    CodeListView->Items->Item[i]->SubItems->Strings[2] = "";
}
```

which is obviously incorrect. One needs to put curly braces around the entire `else` block.

Reply:
I have not noticed that so this bug might be not obvious for me